src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -8272,6 +8272,11 @@ bool Unit::HandleProcTriggerSpell(Unit* victim, uint32 damage, AuraEffect* trigg
                     CastDelayedSpellWithPeriodicAmount(this, trigger_spell_id, SPELL_AURA_PERIODIC_HEAL, basepoints0);
                     return true;
                 }
+                // Greater Heal Refund - Gain 100 mana , when target is full HP
+                if (auraSpellInfo->Id == 37594)
```

Comment:
```diff
-if (auraSpellInfo->Id == 37594)
+if (auraSpellInfo->Id == 37594 && victim->IsFullHealth())
```

Try this, i think this should work ( i am at work so i can't test it in game )

Reply:
ok

Reply:
Does it work now?

Reply:
I did not check the cicumstances, but why is this not in a seperate aura/spell script?